nix/overlay.nix

```diff
@@ -60,9 +60,16 @@ in rec {
   go-modvendor = callPackage ./pkgs/go-modvendor { };
   codecov-cli = callPackage ./pkgs/codecov-cli { };
   go-generate-fast = callPackage ./pkgs/go-generate-fast { };
+
   # brough in gomobile derivation from https://github.com/NixOS/nixpkgs/commit/f5abef98e8b8c9f9e6da4bdab63f8be1e57ea8c0
   # enabled CGO_ENABLED for status-mobile
   # swapped --replace plag with --replace-queit in substituteInPlace block because its deprecated in newer nix versions
   # swapped buildGoModule with buildGo122Module to ensure derivation is built with go 1.22
-  gomobile = callPackage ./pkgs/gomobile {};
-}
+  gomobile = if prev.stdenv.hostPlatform.isDarwin then
+   # CGO compilation fails with newer apple SDK
+    prev.darwin.apple_sdk_11_0.callPackage ./pkgs/gomobile {
+      darwin = prev.darwin;
+    }
+  else
+    callPackage ./pkgs/gomobile { };
+}
```

Comment:
Shouldn't this already be happening by default? Based off of this:
```nix
      # Pick an SDK
      apple_sdk =
        {
          "10.12" = apple_sdk_10_12;
          "11.0" = apple_sdk_11_0;
        }
        .${stdenv.hostPlatform.darwinSdkVersion}
          or (throw "Unsupported sdk: ${stdenv.hostPlatform.darwinSdkVersion}");
```
https://github.com/NixOS/nixpkgs/blob/3939f5f357444b182817a65b86019dbcb16bdb65/pkgs/top-level/darwin-packages.nix#L78-L85

Reply:
correct.